crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -352,6 +359,26 @@ impl Backend {
 
     /// Resets the fork to a fresh state
     pub async fn reset_fork(&self, forking: Forking) -> Result<(), BlockchainError> {
+        if !self.is_fork() {
+            if let Some(eth_rpc_url) = forking.clone().json_rpc_url {
+                let mut env = self.env.read().clone();
+
+                let mut node_config = self.node_config.write().await;
+                let (db, forking) =
+                    node_config.fork_db_setup(eth_rpc_url, &mut env, &self.fees).await;
+
+                *self.env.write() = env;
+                self.db.write().await.init_from_snapshot(db.write().await.clear_into_snapshot());
```

Comment:
I'm pretty sure that this is not going to work if I request something that's not in the local DB already. Trying to replace the `dyn Db` in `self.db` is not going well because it's not sized. I think I'll need to `Box` it, but seeing if I can avoid that.